src/main/java/seedu/contax/ui/onboarding/OnboardingWindow.java

```diff
@@ -119,126 +117,270 @@ public void translate(double x, double y) {
-                persons.add(step.getPerson());
-            } else if (step.getPersonOperation() == 2) {
-                mainWindow.show();
-                stage.hide();
+    private void initStoryManager(OnboardingStoryManager mng) {
+        processStep(storyManager.start());
+        parentPane.setOnMouseClicked((event -> {
+            OnboardingStep s = storyManager.handleMouseClick();
+            if (s != null) {
+                processStep(s);
+                processInstructionPosition(s.getPositionOption());
             }
+        }));
 
-            if (step.getCommand() != null) {
-                if (!commandBox.getText().equals(step.getCommand())) {
-                    if (!commandBox.getText().equals("")) {
-                        instructionLabel.setText("Please type: " + step.getCommand());
-                    }
-                    return;
-                } else {
-                    OnboardingStep s = storyManager.getNextStep();
+        commandBox.getRoot().setOnKeyPressed((event) -> {
+            if (event.getCode() == KeyCode.ENTER) {
+                OnboardingStep s = storyManager.handleEnter();
+                if (s != null) {
                     processStep(s);
-                    step.setEventType(s.getPositionOption());
+                    processInstructionPosition(s.getPositionOption());
                 }
             }
-            storyManager.stepFront();
-        }
-
+        });
     }
 
     /**
-     * Highlight the stage based on the given option
+     * Highlights a node in this window the stage based on the given option
+     * <br><br>
+     * Options are as follows:
+     * <br>- 0: clear all highlights
+     * <br>- 1: highlight only the command box
+     * <br>- 2: highlight only the person list
      * @param option Highlight option
      */
-    public void processHighlightOption(int option) {
-        if (option == 0) {
+    private void processHighlightOption(int option) {
+        switch (option) {
+        case 0:
             commandBox.unhighlight();
             commandBox.clear();
             personList.setStyle("-fx-border-width: 0px");
-        } else if (option == 1) {
+            break;
+        case 1:
             commandBox.highlight();
-        } else if (option == 2) {
+            break;
+        case 2:
             commandBox.unhighlight();
             commandBox.clear();
             personList.setStyle("-fx-border-color: yellow; -fx-border-width: 5px");
+            break;
+        default:
+            break;
         }
     }
 
     /**
-     * Overlay the stage based on the given option
-     * @param option Overlay option
+     * Displays an overlay on this window, whose position is dependant on the given option
+     * <br><br>
+     * Options are as follows:
+     * <br>- 0: cover all
+     * <br>- 1: cover all, showing only the menu bar
+     * <br>- 2: cover all, showing only the command box
+     * <br>- 3: cover all, showing only the person list
```

Comment:
Would be better if you abstract it into an Enum, which can be nested in OnboardingStory if necessary.